sdks/java/core/src/main/java/org/apache/beam/sdk/options/ProxyInvocationHandler.java

```diff
@@ -732,14 +732,22 @@ private void ensureSerializable(
     @Override
     public PipelineOptions deserialize(JsonParser jp, DeserializationContext ctxt)
         throws IOException, JsonProcessingException {
-      ObjectNode objectNode = (ObjectNode) jp.readValueAsTree();
-      ObjectNode optionsNode = (ObjectNode) objectNode.get("options");
+      ObjectNode objectNode = jp.readValueAsTree();
+      JsonNode rawOptionsNode = objectNode.get("options");
+      ObjectNode optionsNode;
+      if (rawOptionsNode == null || rawOptionsNode.isNull()) {
```

Comment:
Can this not be pulled up some? e.g. 

```
PipelineOptions options;
if (rawOptionsNode == null || rawOptionsNode.isNull()) {
  options = PipelineOptionsFactory.create(); // or new ProxyInvocationHandler..., but I believe these to be equivalent
} else {
  // code
  options = ProxyInvocationHandler...
}
ValueProvider.RuntimeValueProvider.setRuntimeOptions(options)
return options;
```

Because the two if statements aren't independent - so we can just glom them together and have a more obvious path through the method.

Reply:
As it turns out, the second if statement is completely redundant. ;)

I wasn't confident about the equivalence between an empty `ProxyInvocationHandler` reconstruction and `PipelineOptionsFactory.create()`, so I've left that as it was.